src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortToPgen.java

```diff
@@ -0,0 +1,272 @@
+    @Argument(
+            fullName = "pgen-chromosome-code",
+            shortName = "pcc",
+            doc = "Plink defines a set of chromosome codes that correspond to different sets of contig names for" +
+                    " chromosomes. This tool supports codes 'chrM' and 'MT'.  Chromosome code reference for plink2" +
+                    " can be found here: https://www.cog-genomics.org/plink/2.0/data#irreg_output"
+    )
+    private ChromosomeCode pgenChromosomeCode;
+
+    @Argument(
+            shortName = "wm",
+            fullName = "write-mode",
+            doc = "Write mode for the PGEN writer.",
+            optional = true
+    )
+    private WriteMode writeMode = WriteMode.WRITE_AND_COPY;
+
+    @Argument(
+            fullName = "max-alt-alleles",
+            shortName = "maa",
+            doc = "Maximum alt alleles per site.  Cannot exceed 254.  Sites with more than the max will be excluded.",
+            maxValue = PgenWriter.PLINK2_MAX_ALTERNATE_ALLELES,
+            optional = true
+    )
+    private int maxAltAlleles = PgenWriter.PLINK2_MAX_ALTERNATE_ALLELES;
+
+    @Argument(
+            fullName = "lenient-ploidy-validation",
+            shortName = "lpv",
+            doc = "PGEN requires individual sample to be diploid (except for sex chromosomes, which may be haploid - " +
+                    "these are accepted and recoded for pgen as heterozygous/diploid). By default, any ploidy failure " +
+                    "will result in an exception to be thrown. Use this flag to tolerate ploidy failures " +
+                    "(samples will be recoded as missing, and logged if a log file is provided using --writerLogFile)",
+            optional = true
+    )
+    private boolean lenientPloidyValidation = false;
+
+    @Argument(
+            fullName = "writer-log-file",
+            shortName = "wlf",
+            doc = "An optional second log file that the PGEN Writer will use to log sites dropped for exceeding " +
+                    "maxAltAlleles, and also, if --lenientPloidyValidation is used, logs sites and samples where " +
+                    "samples have to be marked as missing for non-conforming ploidy.",
+            optional = true
+    )
+    private String writerLogFile = null;
+
+    @Argument(
+            fullName = "allow-empty-pgen",
+            doc = "In the case that no variants are written to the PGEN file, this flag allows the tool to write an " +
+                    "empty PGEN file.  By default, the tool will throw an exception if no variants are written, because" +
+                    " an empty PGEN file is not technically a valid PGEN file.",
+            optional = true
+    )
+    private boolean allowEmptyPgen = false;
+
+    @Argument(
+            fullName = "preserve-phasing",
+            doc = "Preserves phasing in the output PGEN file if phasing is present in the source genotypes.",
+            optional = true
+    )
+    private boolean preservePhasing = false;
```

Comment:
wouldn't we want phasing to default to being on?

Reply:
This is a good question.  It's disabled by default in the PGEN writer library.  I think it maybe makes the files bigger because it changes how the data is stored.  I could change it here, maybe default it to preserve phasing, and have a flag called "--ignore-phasing" or something.  Or I could change it in the WDL to default to preserve phasing.  What do you think?

Reply:
ah the size is something I hadn't thought about (silly me) and that's enough to convince me